File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/JavaClassEntity.java
package com.github.ol_loginov.heaplibweb.repository.heap;

import com.github.ol_loginov.heaplibweb.repository.EntityInstance;
import lombok.*;

@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class JavaClassEntity implements EntityInstance {
	private int heapId;
	private long javaClassId;

	private String name;
	private long allInstancesSize;
	private boolean array;
	private int instanceSize;
	private int instancesCount;
	private long retainedSizeByClass;
	private Long superClassId;
}

File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/JavaClassEntity.kt
package com.github.ol_loginov.heaplibweb.repository.heap

import com.github.ol_loginov.heaplibweb.repository.EntityInstance
import org.springframework.lang.NonNull

class JavaClassEntity(
    var javaClassId: Long,

    var name: String,
    var allInstancesSize: Long?,
    var array: Boolean?,
    var instanceSize: Int,
    var instancesCount: Int?,
    var retainedSizeByClass: Long?,
    var superClassId: Long?
) : EntityInstance {
    internal constructor() : this(0, "", null, null, 0, null, null, null)

    @NonNull
    override fun toString() = "${javaClass.simpleName}#$javaClassId"
}